src/routers/mod.rs

```diff
@@ -1,5 +1,28 @@
+use anyhow::Result;
+use pyo3::{Py, PyAny};
+
 pub mod const_router;
 pub mod middleware_router;
 pub mod router;
 pub mod types;
 pub mod web_socket_router;
+
+pub struct RouteType<T>(pub T);
+
+pub trait Router<T, U> {
+    /// Checks if the functions is an async function
+    /// Inserts them in the router according to their nature(CoRoutine/SyncFunction)
+    fn add_route(
+        &self,
+        route_type: &str,
+        route: &str,
+        handler: Py<PyAny>,
+        is_async: bool,
+        number_of_params: u8,
+        event_loop: Option<&PyAny>,
+    ) -> Result<()>;
+
+    /// Checks if the functions is an async function
+    /// Inserts them in the router according to their nature(CoRoutine/SyncFunction)
```

Comment:
One nit: `get` function won't insert the function in the router.

Reply:
Oops I copied the wrong comment, will fix that